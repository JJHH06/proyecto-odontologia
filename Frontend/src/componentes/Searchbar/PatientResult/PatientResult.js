import {React, useState} from 'react';
import { Link } from 'react-router-dom';
import { useHistory } from 'react-router-dom'

const PatientResult = (props) => {
  const items = props;
  const history = useHistory();

  const handleFicha = () => {
    console.log('si se ejecuta');
    history.push({
      pathname: `/ficha`,
      state: items.paciente 
    });
  }
  
  return (
    <div data-testid="paciente-buscado" className = "row paciente-busqueda" onClick={handleFicha}>
      
                <div className="col">
                        {items.paciente.nombre}
                </div>
                <div className="col">
                {items.paciente.edad}

                </div>

                <div className="col">
                {items.paciente.telefono_celular}
                </div>
                    </div>
  );
};

export default PatientResult;
