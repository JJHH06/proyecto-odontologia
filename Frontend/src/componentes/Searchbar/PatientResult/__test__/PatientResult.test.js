import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {render, cleanup, screen} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import PatientResult from '../PatientResult';

const TESTPATIENT = {edad: 12, nombre: "TestPatient", telefono_celular: "12345678"}


afterEach(cleanup);
it("Renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<PatientResult paciente = {TESTPATIENT}/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("Renders PatientResult Correctly",() =>{
    const {getByTestId}= render(<PatientResult paciente = {TESTPATIENT}/>)
    expect(getByTestId('paciente-buscado')).toHaveTextContent("TestPatient")
});

it('PatientResult results are displayed', async () => {
    render(<PatientResult paciente = {TESTPATIENT}/>);
    expect(screen.getByTestId('paciente-buscado')).toBeInTheDocument();
});

it("Matches snapshot",() =>{
    const tree = renderer.create(<PatientResult paciente = {TESTPATIENT}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
