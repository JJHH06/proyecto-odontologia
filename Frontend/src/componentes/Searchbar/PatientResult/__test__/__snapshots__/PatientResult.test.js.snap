// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="row paciente-busqueda"
  data-testid="paciente-buscado"
  onClick={[Function]}
>
  <div
    className="col"
  >
    TestPatient
  </div>
  <div
    className="col"
  >
    12
  </div>
  <div
    className="col"
  >
    12345678
  </div>
</div>
`;
